src/Nest/CommonAbstractions/Infer/Field/FieldExpressionVisitor.cs

```diff
@@ -14,7 +14,9 @@ namespace Nest
 {
 	internal class HasConstantExpressionVisitor : ExpressionVisitor
 	{
-		public bool Found { get; private set; }
+		public bool Found => this.FoundConstants.Any(b => b);
```

Comment:
Yea, good call. No need to maintain a collection at all, as long as we still visit each method call then we can use just the `Found` flag.  I've changed this with my last commit.